### PR TITLE
Start PostgreSQL on Windows via TCP on dedicated port.

### PR DESCRIPTION
**Pull request by nelsonjr**
## Test plan

Windows testing is currently manually executing the application and confirms it starts.

### CODE REVIEW

**Review by peterguy — APPROVED** [review]
LGTM!

**Review by chwarwick — DISMISSED** [review]
@nelsonjr  The scope of this change effects more then just the Windows build, and the included test plan is entirely insufficient.  

Reliability and stability of the embedded postgres is an issue that the app team has spent a considerable amount of time on during this cycle and if we **need** to update with approximately 48 hours to the event it should be more throughly tested and the change should be more widely publicized.

Additionally these changes break the non-windows build.

cc @slimsag

File: internal/singleprogram/postgresql.go [hunk 1]
```diff
@@ -115,6 +138,8 @@ func startEmbeddedPostgreSQL(logger log.Logger, pgRootDir string) (StopPostgresF
 			Username(vars.PGUSER).
 			Database(vars.PGDATABASE).
 			UseUnixSocket(unixSocketDir).
+			Port(pgPort).
+			Password(vars.PGPASSWORD).
```
Comment (slimsag) [inline internal/singleprogram/postgresql.go:142]: Can we isolate this change to just Windows?

File: internal/singleprogram/singleprogram.go [hunk 2]
```diff
@@ -37,6 +37,10 @@ func Init(logger log.Logger) CleanupFunc {
 	// INDEXED_SEARCH_SERVERS is empty (but defined) so that indexed search is disabled.
 	setDefaultEnv(logger, "INDEXED_SEARCH_SERVERS", "")
 
+	// POSTGRES database, specifying a non-default port to avoid conflicting with developer's
+	// local servers, if they happen to have PostgreSQL running on their machines.
+	setDefaultEnv(logger, "PGPORT", "5434")
```
Comment (slimsag) [inline internal/singleprogram/singleprogram.go:42]: Can we isolate this change to just Windows?

File: src-tauri/tauri.conf.json [hunk 3]
```diff
@@ -48,7 +48,7 @@
       "deb": {
         "depends": []
       },
-      "externalBin": ["../.bin/sourcegraph-backend"],
+      "externalBin": ["../../.bin/sourcegraph-backend"],
```
Comment (slimsag) [inline src-tauri/tauri.conf.json:51]: I think this will break building for non-Windows platforms. Why can't the binary be in the same location?

**Comment by slimsag** [comment]
Can you make sure the mac builds still work after this change? I'm fine with this going in but let's make sure it's very well tested since this is pretty risky to merge, especially the changes around postgres port/password

**Comment by nelsonjr** [comment]
> Can you make sure the mac builds still work after this change? I'm fine with this going in but let's make sure it's very well tested since this is pretty risky to merge, especially the changes around postgres port/password

Good catch. Will do!

File: internal/singleprogram/singleprogram.go (hunk 2 above)
Comment (nelsonjr) [reply internal/singleprogram/singleprogram.go:42]: Ditto.

File: internal/singleprogram/postgresql.go (hunk 1 above)
Comment (nelsonjr) [reply internal/singleprogram/postgresql.go:142]: They are per: https://github.com/sourcegraph/embedded-postgres/blob/8f62b6947f2c0959c9b8d23911c7ad2dddfde0d8/prepare_database.go#L57-L73

**Comment by slimsag** [comment]
I think myself (and Chris probably) would feel a lot more comfortable with this change if `PGPORT`, `Port(pgPort)`, and `Password(vars.PGPASSWORD)` were behind a windows conditional. I understand that technically they already should be as a result of the embedded-postgres code side, but I'm not sure I feel confident in vetting that right now or if there are other consequences/side-effects (on e.g. other Sourcegraph services) of setting `PGPORT` in the environment.

I'd be comfortable removing those after the release this week, but I think the most safe thing would be to put those behind a `if windows` for now.

Ultimately I defer to your discretion here, I trust you - that's just my advice for the safest way to proceed given we're less than 48 hours away from the release

File: src-tauri/tauri.conf.json (hunk 3 above)
Comment (slimsag) [reply src-tauri/tauri.conf.json:51]: @nelsonjr last issue here

**Comment by slimsag** [comment]
Looks great to me aside from https://github.com/sourcegraph/sourcegraph/pull/54103/files#r1242348818

File: go.mod [hunk 4]
```diff
@@ -10,7 +10,7 @@ replace (
 	// We maintain a potentially long-term fork of embedded-postgres for
 	// Sourcegraph App. We add features like unix sockets. If this replace
 	// directive still exists in 2024, lets consider hard forking.
-	github.com/fergusstrange/embedded-postgres => github.com/sourcegraph/embedded-postgres v1.19.1-0.20230313141935-8f62b6947f2c
+	github.com/fergusstrange/embedded-postgres => github.com/sourcegraph/embedded-postgres v1.19.1-0.20230624001757-345a8df15ded
```
Comment (keegancsmith) [inline go.mod:13]: minor: unlikely affecting your testing, but for anything built via bazel (IE our CI artifacts) any change to go.mod needs a corresponding change to deps.bzl. You can update it via running `bazel run :gazelle-update-repos`

File: src-tauri/tauri.conf.json (hunk 3 above)
Comment (nelsonjr) [reply src-tauri/tauri.conf.json]: Reverted. @slimsag PTAL.

File: go.mod (hunk 4 above)
Comment (nelsonjr) [reply go.mod:13]: Thank you! Updated.

**Review by slimsag — APPROVED** [review]
Looks great, thank you @nelsonjr !

**Comment by slimsag** [comment]
(I'm satisfied these changes are isolated enough now that you don't need to create a macos build; as long as CI passes)

**Comment by peterguy** [comment]
> PGPORT, Port(pgPort), and Password(vars.PGPASSWORD) were behind a windows conditional.
Might be a good idea to separate into `postgresql.go` and `postgresql_windows.go` to automatically separate into builds for different targets.

**Comment by nelsonjr** [comment]
> > PGPORT, Port(pgPort), and Password(vars.PGPASSWORD) were behind a windows conditional.
> > Might be a good idea to separate into `postgresql.go` and `postgresql_windows.go` to automatically separate into builds for different targets.

Good idea. I will split them on a separate MR, as there are other functions we don't want to duplicate.

File: .gitignore [hunk 5]
```diff
@@ -77,6 +77,7 @@ pnpm-debug.log
 /ui/assets/*.br
 /ui/assets/*.gz
 /ui/assets/stats-*
+/ui/assets/enterprise/
```
Comment (jhchabran) [inline .gitignore:80]: @nelsonjr why did we add this one to the `.gitignore`? It contains Go code and a buildfile. Are you generating the bundle with bazel or with pnpm?

Comment (nelsonjr) [reply .gitignore:80]: It combines Bazel, pnpm, MSBuild (Visual Studio), and some post build steps. There are two scripts to build on windows (that later we can try do "bazelify"... `build-windows.sh` and `build.ps1`.

**Pull request merged**